src/app/app.component.ts

```diff
@@ -96,6 +98,7 @@ export class AppComponent {
       // ensure deployment field set correctly for use in any startup services or templates
       localStorage.setItem(this.appFields.DEPLOYMENT_NAME, this.DEPLOYMENT_NAME);
       localStorage.setItem(this.appFields.APP_VERSION, this.APP_VERSION);
+      await this.checkForAppUpdatesAndApply();
```

Comment:
nit(blocking)
Would be better to move all init code to service itself. If it is actually required to run before anything else then you can put it first in the list of `blocking` services below (although I expect this is actually something that can be deferred to background and run async)